interfaces/builtin/fwupd.go

```diff
@@ -65,17 +65,26 @@ const fwupdPermanentSlotAppArmor = `
   # Allow fwupd to access system information
   /sys/devices/virtual/dmi/id/product_name r,
   /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/ r,
+  /sys/firmware/dmi/tables/** r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
   # Allow write access for efi firmware updater
+  /boot/efi/{,**/} r,
+  /boot/efi/EFI/ubuntu/fwupd*.efi* rw,
   /boot/efi/EFI/ubuntu/fw/** rw,
 
   # Allow access from efivar library
   owner @{PROC}/@{pid}/mounts r,
+  /sys/dev/block/**/partition r,
```

Comment:
The files in /sys/dev/block are all symlinks to /sys/devices/... so this rule should instead be: `/sys/devices/**/partition r,` (but see below about just plugging hardware-observe). Can you paste a denial and the system it came from that prompted this rule if you feel this assessment is in error?

Reply:
Yes, they can be removed by using hardware-observe. That's great cleanup for keeping simple for fwupd if!

Reply:
It sounds like hardware-observe does handle a majority of read cases, but let me list out a few stlil missing:
* for nvme FW update it will need to perform ioctl to `/dev/nvme*`
* For TBT3/USB4/TBT Retimer update will need to write to several files in `/sys/bus/thunderbolt/`
* For DP aux based update such as MST update will need to read and write to `/dev/drm_dp_aux*`
* For USB based update will need `/dev/bus/usb/*`